File: rest-api/app/Models/Restaurant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Restaurant extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'description',
        'grade',
        'localization',
        'phone_number',
        'website',
        'hours'
    ];

    public static function getMean($id)
    {
        $restaurant = Restaurant::find($id);
        if ($restaurant == null) {
            return 0;
        }
        $reviews = $restaurant->reviews;
        if (!$reviews->count()) {
            return $restaurant->grade;
        }
        $sum = 0;
        $length = 0;
        $reviews->each(function (Review $review) use (&$length, &$sum) {
            $sum += $review->grade;
            $length += 1;
        });
        return round($sum / $length, 2);
    }

    public function menus()
    {
        return $this->hasMany(Menu::class, "restaurant_id");
    }

    public function reviews()
    {
        return $this->hasMany(Review::class, "restaurant_id");
    }

    protected $collection = "restaurants";
}

File: four-metal/app/src/main/java/com/tah/fourmetal/data/models/Restaurant.kt
package com.tah.fourmetal.data.models

import android.os.Parcel
import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.io.Serializable

@Parcelize
data class Restaurant(
    var id: Int,
    var name: String?,
    var description: String?,
    var grade: Float,
    var image_url:String?,
    var localization: String?,
    var phone_number: String?,
    var website: String?,
    var hours: String?
) : Parcelable